packages/parcel-resolver-post/src/handle-ts-path.ts

```diff
@@ -169,25 +180,61 @@ function attemptResolveArray(
   return null
 }
 
-async function getTsconfigCompilerOptions({
-  options,
-  dependency
-}: ResolverProps) {
+type TSConfig = { compilerOptions: CompilerOptions; filePath: string }
+
+async function getTsconfigCompilerOptions(
+  props: ResolverProps & { tsconfigPath?: string },
+  tsConfigs: TSConfig[] = [],
+  depth = 0
+): Promise<TSConfig[]> {
+  if (depth > 42) {
+    throw new Error(
+      "Something went wrong in loading tsconfig (depth > 42). Circular dependency?"
+    )
+  }
+
+  const { options, dependency, tsconfigPath } = props
+
+  const tsconfigPathList = tsconfigPath
+    ? [tsconfigPath]
+    : ["tsconfig.json", "tsconfig.js"]
+
   const result = await loadConfig(
     options.inputFS,
     dependency.resolveFrom,
-    ["tsconfig.json", "tsconfig.js"],
+    tsconfigPathList,
     join(process.env.PLASMO_PROJECT_DIR, "lab")
   )
 
-  if (!result?.config?.compilerOptions) {
-    return null
+  const compilerOptions = result?.config?.compilerOptions as CompilerOptions
+
+  if (!compilerOptions) {
+    return tsConfigs
```

Comment:
Since we're doing recursion, this should return the passed down data otherwise we would be missing result.